Learn/Lessons/TimeInRangeLesson.swift

```diff
@@ -0,0 +1,265 @@
+//
+//  LessonPlayground.swift
+//  Learn
+//
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopCore
+import LoopKit
+import LoopKitUI
+import LoopUI
+import HealthKit
+import os.log
+
+
+final class TimeInRangeLesson: Lesson {
+    let title = NSLocalizedString("Time in Range", comment: "Lesson title")
+
+    let subtitle = NSLocalizedString("Computes the percentage of glucose measurements within a specified range", comment: "Lesson subtitle")
+
+    let configurationSections: [LessonSectionProviding]
+
+    private let dataManager: DataManager
+
+    private let glucoseUnit: HKUnit
+
+    private let glucoseFormatter = QuantityFormatter()
+
+    private let dateEntry: DateEntry
+
+    private let weeksEntry: NumberEntry
+
+    private let rangeEntry: QuantityRangeEntry
+
+    init(dataManager: DataManager) {
+        self.dataManager = dataManager
+        self.glucoseUnit = dataManager.glucoseStore.preferredUnit ?? .milligramsPerDeciliter
+
+        let twoWeeksAgo = Calendar.current.date(byAdding: DateComponents(weekOfYear: -2), to: Date())!
+
+        glucoseFormatter.setPreferredNumberFormatter(for: glucoseUnit)
+
+        // TODO: Add a date components picker cell, and combine into a "DateIntervalEntry" section
+        dateEntry = DateEntry(
+            date: Calendar.current.startOfDay(for: twoWeeksAgo),
+            title: NSLocalizedString("Start Date", comment: "Title of config entry"),
+            mode: .date
+        )
+        weeksEntry = NumberEntry.integerEntry(
+            value: 2,
+            unitString: NSLocalizedString("Weeks", comment: "Unit string for a count of calendar weeks")
+        )
+
+        rangeEntry = QuantityRangeEntry.glucoseRange(quantityFormatter: glucoseFormatter, unit: glucoseUnit)
+
+        self.configurationSections = [
+            LessonSection(headerTitle: nil, footerTitle: nil, cells: [dateEntry, weeksEntry]),
+            rangeEntry
+        ]
+    }
+
+    func execute(completion: @escaping ([LessonSectionProviding]) -> Void) {
+        guard let weeks = weeksEntry.number?.intValue, let closedRange = rangeEntry.closedRange else {
+            // TODO: Cleaner error presentation
```

Comment:
TODO?

Reply:
Yeah, this probably deserves a larger thinking out of input validation, or how to make the inputs not allow erroneous input.